### PR TITLE
Add Fulfillment Service

### PR DESCRIPTION
**Pull request by gir**
Creating for @aryy this PR adds Fulfillment and helpers to Order.

### CODE REVIEW

File: filfillment_test.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
+package goshopify
```
Comment (dlebech) [inline filfillment_test.go]: There is a typo in the filename

File: fulfillment.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	wrappedData := FulfillmentResource{Fulfillment: &fulfillment}
+	resource := new(FulfillmentResource)
+	err := s.client.Put(path, wrappedData, resource)
+	return resource.Fulfillment, err
+}
+
+// Complete an existing fulfillment
+func (s *FulfillmentServiceOp) Complete(fulfillmentID int) (*Fulfillment, error) {
+	prefix := FulfillmentPathPrefix(s.resource, s.resourceID)
+	path := fmt.Sprintf("%s/%d.json", prefix, fulfillmentID)
```
Comment (dlebech) [inline fulfillment.go]: I think this needs to be:
`path := fmt.Sprintf("%s/%d/complete.json", prefix, fulfillmentID)`

File: fulfillment.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	resource := new(FulfillmentResource)
+	err := s.client.Post(path, nil, resource)
+	return resource.Fulfillment, err
+}
+
+// Cancel an existing fulfillment
+func (s *FulfillmentServiceOp) Cancel(fulfillmentID int) (*Fulfillment, error) {
+	prefix := FulfillmentPathPrefix(s.resource, s.resourceID)
+	path := fmt.Sprintf("%s/%d.json", prefix, fulfillmentID)
```
Comment (dlebech) [inline fulfillment.go]: I think this needs to be:
`path := fmt.Sprintf("%s/%d/cancel.json", prefix, fulfillmentID)`

File: fulfillment.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	resource := new(FulfillmentResource)
+	err := s.client.Post(path, nil, resource)
+	return resource.Fulfillment, err
+}
+
+// Transition an existing fulfillment
+func (s *FulfillmentServiceOp) Transition(fulfillmentID int) (*Fulfillment, error) {
+	prefix := FulfillmentPathPrefix(s.resource, s.resourceID)
+	path := fmt.Sprintf("%s/%d.json", prefix, fulfillmentID)
```
Comment (dlebech) [inline fulfillment.go]: I think this needs to be:
`path := fmt.Sprintf("%s/%d/open.json", prefix, fulfillmentID)`

**Review by dlebech — CHANGES_REQUESTED** [review]
Thanks for PR @gir There seems to be a few typos, but otherwise looks ok :-)

**Comment by andrewhoff** [comment]
Hey @dlebech we're going to pull this in here, then make some changes on our fork

**Review by dlebech — APPROVED** [review]

**Comment by dlebech** [comment]
@andrewhoff yeah no problem. Sorry for the wait. Feel free to merge when you're ready, the changes look good to me :-)

Thanks for the changes @gir 👍

**Pull request merged**